atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/timestamp/ConnectionDbTypes.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2015 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.dbkvs.timestamp;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.SQLException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.palantir.nexus.db.DBType;
+
+public class ConnectionDbTypes {
+
+    private static final Logger log = LoggerFactory.getLogger(ConnectionDbTypes.class);
+
+    public static DBType getDBType(Connection conn) {
```

Comment:
nit: This is kind of a funky class.  It's only used in once in OSS and only two times in the internal use case.  Also it is really a util class so should either be combined with another or properly labeled as such and made non-instantiatable.